### PR TITLE
Add test to ensure consistent constraints in parallel

### PR DESCRIPTION
**Pull request by kronbichler**
This is the test needed for #14905. I added it as a separate PR as it can currently run (but would fail to produce a reasonable output). The idea of the test is to set up constraints for a DoF owned at process with index 0, but defined only from information on processes with rank 1 and 2. The problem to be represented is that neither of process 1 and 2 have the full information on the constraints, and only their combined form represents the valid state.

### CODE REVIEW

**Review by bangerth — APPROVED** [review]
This is ok to merge. Out of curiosity, what is the situation you were considering that triggered this whole thing?

**Comment by bangerth** [comment]
(Ok to merge = once #14905 is merged and this is rebased.)

File: tests/mpi/constraints_consistent_02.mpirun=3.output
Comment (peterrum) [inline tests/mpi/constraints_consistent_02.mpirun=3.output:1]: Is the output already generated with https://github.com/dealii/dealii/pull/14905?

Comment (kronbichler) [reply tests/mpi/constraints_consistent_02.mpirun=3.output:1]: Yes, this uses the code of #14905.

**Review by peterrum — COMMENTED** [review]
@kronbichler Could you rename the file to `tests/lac/constraints_make_consistent_in_parallel_02.cc` so that the `AC::make_consistent_in_parallel()` are at the same place!?

File: tests/mpi/constraints_consistent_02.mpirun=3.output
Comment (peterrum) [reply tests/mpi/constraints_consistent_02.mpirun=3.output:1]: Are you sure that the output is up to date? Where is `Working on rank` coming from and why do not all processes output `Output 0`, `Output 1`, and `Output 2`?

Comment (peterrum) [reply tests/mpi/constraints_consistent_02.mpirun=3.output:1]: The output should be:
```

DEAL:0::Output 0
DEAL:0::Output 1
    0 1:  0.500000
    0 3:  0.500000
    0 2:  0.500000
    2 3:  1.50000
DEAL:0::Output 2
    0 1:  0.500000
    0 3:  1.25000
    2 3:  1.50000

DEAL:1::Output 0
    0 1:  0.500000
    0 2:  0.500000
    2 3:  1.50000
DEAL:1::Output 1
    0 1:  0.500000
    0 3:  0.500000
    0 2:  0.500000
    2 3:  1.50000
DEAL:1::Output 2
    0 1:  0.500000
    0 3:  1.25000
    2 3:  1.50000


DEAL:2::Output 0
    0 1:  0.500000
    0 3:  0.500000
DEAL:2::Output 1
    0 1:  0.500000
    0 3:  0.500000
    0 2:  0.500000
    2 3:  1.50000
DEAL:2::Output 2
    0 1:  0.500000
    0 3:  1.25000
    2 3:  1.50000

```

**Comment by kronbichler** [comment]
> Could you rename the file to tests/lac/constraints_make_consistent_in_parallel_02.cc so that the AC::make_consistent_in_parallel() are at the same place!?

I will adapt the output once #14905 is merged. Regarding the name, I used the `mpi/constraints_consistent_01.cc` as a template, so we have a conflict. I can move that file to the `lac` subfolder, too, as it feels that all the three tests should be in the same place.

**Comment by kronbichler** [comment]
> Could you rename the file to tests/lac/constraints_make_consistent_in_parallel_02.cc so that the AC::make_consistent_in_parallel() are at the same place!?

I will adapt the output once #14905 is merged. Regarding the name, I used the `mpi/constraints_consistent_01.cc` as a template, so we have a conflict. I can move that file to the `lac` subfolder, too, as it feels that all the three tests should be in the same place.

**Comment by kronbichler** [comment]
This was integrated into #14905.

**Pull request closed without merging**
